frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -162,6 +169,41 @@ function isUserActivity(snapshot: eventWithTime): boolean {
     )
 }
 
+const updatePlayerTimeTracking = (
+    current: PlayerTimeTracking,
+    newState: PlayerTimeTracking['state']
+): PlayerTimeTracking => {
+    // if we were just playing then update watch time
+    const newWatchTime =
+        current.lastTimestamp !== null && current.state === 'playing'
+            ? current.watchTime + (performance.now() - current.lastTimestamp)
+            : current.watchTime
+
+    // if we were just buffering then update buffer time
+    const newBufferTime =
+        current.lastTimestamp !== null && current.state === 'buffering'
+            ? current.bufferTime + (performance.now() - current.lastTimestamp)
+            : current.bufferTime
```

Comment:
this logic was spread around and difficult to follow because we were tracking two different states

but, much simpler, if we say...

* if the last state was playing then we should increment the watch time
* if the last state was buffering then we should increment the buffering time